mono/metadata/object.c

```diff
@@ -7222,20 +7193,11 @@ mono_unichar4*
 mono_string_to_utf32 (MonoString *s)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
-
-	mono_unichar4 *utf32_output = NULL; 
-	GError *error = NULL;
-	glong items_written;
 	
 	if (s == NULL)
 		return NULL;
-		
-	utf32_output = g_utf16_to_ucs4 (s->chars, s->length, NULL, &items_written, &error);
-	
-	if (error)
-		g_error_free (error);
 
-	return utf32_output;
+	return g_utf16_to_ucs4 (s->chars, s->length, NULL, NULL, NULL);
```

Comment:
Would you mind also marking `mono_string_utf32` with `MONO_RT_EXTERNAL_ONLY` in `object.h`?
I can't image we'd ever call this in the runtime, but if we do, we should get the compiler to remind us to write a version that doesn't swallow conversion errors.

Reply:
I think it is, because I went going to remove all utf32 code, and found just the embedding api.. I kinda think we should segregate embedded-only in separate files, at least .c, and .h if we can rename upon build/install (don't want to break public file names). Anyway I'll check.

Reply:
> Anyway I'll check.

I checked - it's unused.  Tag it with `MONO_RT_EXTERNAL_ONLY` in the header.  (Actually it can even be a separate / followup PR).

> I kinda think we should segregate embedded-only in separate files, at least .c, and .h if we can rename upon build/install (don't want to break public file names)

Agreed. (Long term; not as part of this PR).